source/common/network/win32_socket_handle_impl.cc

```diff
@@ -0,0 +1,105 @@
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::write(Buffer::Instance& buffer) {
+  Api::IoCallUint64Result result = IoSocketHandleImpl::write(buffer);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Write);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::sendmsg(const Buffer::RawSlice* slices,
+                                                       uint64_t num_slice, int flags,
+                                                       const Address::Ip* self_ip,
+                                                       const Address::Instance& peer_address) {
+
+  Api::IoCallUint64Result result =
+      IoSocketHandleImpl::sendmsg(slices, num_slice, flags, self_ip, peer_address);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Write);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recvmsg(Buffer::RawSlice* slices,
+                                                       const uint64_t num_slice, uint32_t self_port,
+                                                       RecvMsgOutput& output) {
+  Api::IoCallUint64Result result =
+      IoSocketHandleImpl::recvmsg(slices, num_slice, self_port, output);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recvmmsg(RawSliceArrays& slices, uint32_t self_port,
+                                                        RecvMsgOutput& output) {
+  Api::IoCallUint64Result result = IoSocketHandleImpl::recvmmsg(slices, self_port, output);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recv(void* buffer, size_t length, int flags) {
+
+  Api::IoCallUint64Result result = IoSocketHandleImpl::recv(buffer, length, flags);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+IoHandlePtr Win32SocketHandleImpl::accept(struct sockaddr* addr, socklen_t* addrlen) {
+  auto result = Api::OsSysCallsSingleton::get().accept(fd_, addr, addrlen);
+  if (SOCKET_INVALID(result.return_value_)) {
+    return nullptr;
+  }
+
+  return std::make_unique<Win32SocketHandleImpl>(result.return_value_, socket_v6only_, domain_);
+}
+
+IoHandlePtr Win32SocketHandleImpl::duplicate() {
+  auto result = Api::OsSysCallsSingleton::get().duplicate(fd_);
+  RELEASE_ASSERT(result.return_value_ != -1,
+                 fmt::format("duplicate failed for '{}': ({}) {}", fd_, result.errno_,
+                             errorDetails(result.errno_)));
+  return std::make_unique<Win32SocketHandleImpl>(result.return_value_, socket_v6only_, domain_);
+}
+
+void Win32SocketHandleImpl::reEnableEventBasedOnIOResult(const Api::IoCallUint64Result& result,
+                                                         uint32_t event) {
+  if (result.wouldBlock() && file_event_) {
+    file_event_->registerEventIfEmulatedEdge(event);
```

Comment:
I am waiting for this because we don't have any changes lined up for this file so I don't want to churn (refactor) code that is working so far. I am doing this change because I want to add the peak functionality in this class.

If we have another change in `FileEventImpl` I think we could think about splitting it